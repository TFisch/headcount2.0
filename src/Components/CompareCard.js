import React from 'react';
import '../styling/CompareCard.css';
import PropTypes from 'prop-types';

const CompareCard = ({ locationOne, locationTwo, foundAverages }) => {
  const averageKeys = Object.values(foundAverages);

  return (
    <div className="compare-card">
      <p className="location-heading">{locationOne.location}</p>
      <p className="data">{averageKeys[0]}</p>
      <h4 className="location-heading heading-two compare-head">COMPARE</h4>
      <p className="data">{averageKeys[2]}</p>
      <p className="location-heading heading-two">{locationTwo.location}</p>
      <p className="data">{averageKeys[1]}</p>
    </div>
  );
};

CompareCard.propTypes = {
  locationOne: PropTypes.string,
  locationTwo: PropTypes.string,
  foundAverages: PropTypes.array
};

export default CompareCard;
