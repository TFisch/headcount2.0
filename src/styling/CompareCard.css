.compare-card {
  background-color: rgba(237, 245, 225, 0.829);
  border: 2px solid rgb(214, 215, 216);
  cursor: pointer;
  font-family: 'Cabin', sans-serif;
  height: 18em;
  position: relative;
  -webkit-box-shadow: 0 1px 4px rgba(0, 0, 0, 0.3),
    0 0 40px rgba(0, 0, 0, 0.1) inset;
  -moz-box-shadow: 0 1px 4px rgba(0, 0, 0, 0.3),
    0 0 40px rgba(0, 0, 0, 0.1) inset;
  box-shadow: 0 1px 4px rgba(0, 0, 0, 0.3), 0 0 40px rgba(0, 0, 0, 0.1) inset;
  margin: 0.85em;
  padding-bottom: 1em;
  text-align: center;
  width: 15em;
}

.compare-head {
  color: rgba(247, 243, 49, 0.829);
}

.data {
  font-size: 1.2em;
}

.heading-two {
  margin-top: 2em;
}

.location-heading {
  background-color: #05386b;
  border: 2px solid rgb(214, 215, 216);
  color: rgb(46, 182, 107);
  font-family: 'Cabin', sans-serif;
  font-size: 1.1em;
  margin: auto 0;
}
