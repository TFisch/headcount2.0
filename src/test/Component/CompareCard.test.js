import React from 'react';
import { shallow } from 'enzyme';
import CompareCard from '../../Components/CompareCard';

describe('CompareCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <CompareCard
        locationOne={'locationOne'}
        locationTwo={'locationTwo'}
        foundAverages={jest.fn()}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
