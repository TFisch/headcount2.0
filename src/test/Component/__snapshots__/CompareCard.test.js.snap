// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareCard
    foundAverages={[Function]}
    locationOne="locationOne"
    locationTwo="locationTwo"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="location-heading"
/>,
        <p
          className="data"
>
          true
</p>,
        <h4
          className="location-heading heading-two compare-head"
>
          DISTRICTS COMPARED
</h4>,
        <p
          className="data"
>
          Object {
                    "calls": Array [],
                    "instances": Array [],
                  }
</p>,
        <p
          className="location-heading heading-two"
/>,
        <p
          className="data"
>
          [Function]
</p>,
      ],
      "className": "compare-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "location-heading",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": true,
          "className": "data",
        },
        "ref": null,
        "rendered": true,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "DISTRICTS COMPARED",
          "className": "location-heading heading-two compare-head",
        },
        "ref": null,
        "rendered": "DISTRICTS COMPARED",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Object {
            "calls": Array [],
            "instances": Array [],
          },
          "className": "data",
        },
        "ref": null,
        "rendered": Object {
          "calls": Array [],
          "instances": Array [],
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "location-heading heading-two",
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": [Function],
          "className": "data",
        },
        "ref": null,
        "rendered": [Function],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="location-heading"
/>,
          <p
            className="data"
>
            true
</p>,
          <h4
            className="location-heading heading-two compare-head"
>
            DISTRICTS COMPARED
</h4>,
          <p
            className="data"
>
            Object {
                        "calls": Array [],
                        "instances": Array [],
                      }
</p>,
          <p
            className="location-heading heading-two"
/>,
          <p
            className="data"
>
            [Function]
</p>,
        ],
        "className": "compare-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "location-heading",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": true,
            "className": "data",
          },
          "ref": null,
          "rendered": true,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "DISTRICTS COMPARED",
            "className": "location-heading heading-two compare-head",
          },
          "ref": null,
          "rendered": "DISTRICTS COMPARED",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Object {
              "calls": Array [],
              "instances": Array [],
            },
            "className": "data",
          },
          "ref": null,
          "rendered": Object {
            "calls": Array [],
            "instances": Array [],
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "location-heading heading-two",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": [Function],
            "className": "data",
          },
          "ref": null,
          "rendered": [Function],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
